### PR TITLE
[Reader] Add subscribed sites and followed tags to analytics

### LINKED ISSUE #19836: [Reader IA] Add number of followed sites and tags to Reader analytics events
We should add the number of followed sites and tags to Reader analytics events where it makes sense. This should be aligned with iOS so we use the same keys and values.

### PR DESCRIPTION
**Pull request by daniloercoli**
Fixes #19836

-----

## To Test:

* Install Jetpack and sign-in
* Open Reader
* 🔍 Tags followed are correctly tracked
* 🔍 Sites subscribed analytics are tracked

To check analytics you can use the Track Live view tool, use logcat and filter by Reader events, or set few breakpoints in the Reader Tracker.

Note: The FollowedTagsFetched event is triggered multiple times during the normal operation of the Reader, a known issue. To avoid excessive analytics tracking, I've implemented a 1-hour guard. While not ideal—since we might want to track changes in the number of tags and only update when there's a difference—this approach may miss tracking for users who don't often change their tag count, especially if the original tag-followed event occurs outside the analysed timespan. Nonetheless, I'm open to suggestions, though I believe this is a reasonable compromise without having to track both the last event time, and the number of blogs subscribed to in the app settings.

-----

## PR Submission Checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist:

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by RenanLukas** [comment]
Thanks for the PR, @daniloercoli . The followed tags is working as expected. I'm getting this in the logcat:

> 🔵 Tracked: reader_following_fetched, Properties: {"count":"3","type":"tags"}

Before reviewing the code I wanted to ask about the `reader_following_fetched` blogs event: it seems like it's not being tracked when we have `0` blogs. I suppose the `FollowedBlogsChanged` event is not triggered when user has `0` blogs? I wanted to confirm we really don't want to track the `0`.

Besides that, I'm not understanding where the number of blogs is coming from. In the blogs chip and bottom sheet list I'm getting **14** blogs. When I tap "Edit" in the bottom sheet I'm getting **48** blogs and in the `reader_following_fetched` event it's **57**. I think we had a discussion about this number in the past, but I wasn't able to find it. Perhaps this is being tracked in another issue.

According to web it's really **57**, so something's probably off with the Android Reader (not related to your PR)
image

**Comment by daniloercoli** [comment]
> Besides that, I'm not understanding where the number of blogs is coming from. In the blogs chip and bottom sheet list I'm getting **14** blogs. When I tap "Edit" in the bottom sheet I'm getting **48** blogs and in the `reader_following_fetched` event it's **57**. I think we had a discussion about this number in the past, but I wasn't able to find it. Perhaps this is being tracked in another issue.
> 
> According to web it's really **57**, so something's probably off with the Android Reader (not related to your PR)

The value '57' represents the total number of sites you're subscribed to, including P2s, A8C internal sites, and RSS feeds, aligning with what you see on the web. 
It seems the list of sites displayed on Android both on the bottom sheet is missing a subset of your subscribed sites. 

On the Web reader, a call to `https://public-api.wordpress.com/rest/v1.1/read/subscriptions-count` retrieves the total subscriptions, including comments. 
```
{
    "code": 200,
    "headers": [
        {
            "name": "Content-Type",
            "value": "application\/json"
        }
    ],
    "body": {
        "blogs": 302,
        "comments": 2096,
        "pending": 0
    }
}
```


The total number of sites subscribed is also returned at the bottom of the JSON document from `https://public-api.wordpress.com/rest/v1.2/read/following/mine?number=100&page=1`, in the `total_subscriptions ` property, which matches the value from the other endpoint. 

There appears to be an issue with the way the Filter bottom sheet works on the Android Reader, in addition to the already reported pagination issue. However, i think that the ticket here https://github.com/wordpress-mobile/WordPress-Android/issues/20218 does recap both of the problems.

**Comment by daniloercoli** [comment]
> Before reviewing the code I wanted to ask about the `reader_following_fetched` blogs event: it seems like it's not being tracked when we have `0` blogs. I suppose the `FollowedBlogsChanged` event is not triggered when user has `0` blogs? I wanted to confirm we really don't want to track the `0`.

Good point! Actually Subscribed sites analytics are tracked only if there's an addition or deletion, as indicated by if `(!localBlogs.isSameList(serverBlogs)) {`. 
Perhaps we should consider setting a timer that by-passes the count check, similar to what we implemented for the other event, to ensure bumping analytics for people that doesn't change the list often?

**Comment by RenanLukas** [comment]
> Perhaps we should consider setting a timer that by-passes the count check, similar to what we implemented for the other event, to ensure bumping analytics for people that doesn't change the list often?

@daniloercoli makes sense to me 👍  should we have a separate issue and PR for that?

**Comment by daniloercoli** [comment]
> > Perhaps we should consider setting a timer that by-passes the count check, similar to what we implemented for the other event, to ensure bumping analytics for people that doesn't change the list often?
> 
> @daniloercoli makes sense to me 👍 should we have a separate issue and PR for that?

@RenanLukas  Yes, it would probably be better to include the improvements in a separate PR.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 1]
```diff
@@ -948,6 +949,15 @@ public void onEventMainThread(FollowedTagsFetched event) {
...
         }
+
+        // Check last time we've bumped tags followed analytics for this user,
+        // and bumping again if > 1 hrs
+        long tagsUpdatedTimestamp = AppPrefs.getReaderAnalyticsCountTagsTimestamp();
+        long now = new DateProvider().getCurrentDate().getTime();
+        if (now - tagsUpdatedTimestamp > 1000 * 60 * 60) { // 1 hr
+            ReaderTracker.trackFollowedTagsCount(ReaderTagTable.getFollowedTags().size());
+            AppPrefs.setReaderAnalyticsCountTagsTimestamp(now);
+        }
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java:960]: 💡 Would it make sense to move this logic to `ReaderPostListViewModel` so we can unit test this and keep this logic away from the View? The `ViewModel` even has an instance of the `ReaderTracker` injected so we could even avoid the static call.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 2]
```diff
@@ -959,6 +969,8 @@ && hasCurrentTag()
             && (getCurrentTag().isFollowedSites() || getCurrentTag().isDefaultInMemoryTag())) {
             refreshPosts();
         }
+
+        ReaderTracker.trackSubscribedSitesCount(event.getTotalSubscriptions());
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java:973]: 💡 Same question about moving it to the `ViewModel` so it's testable.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/tracker/ReaderTracker.kt [hunk 3]
```diff
@@ -465,6 +466,23 @@ class ReaderTracker @Inject constructor(
...
         }
 
+        private fun trackFollowedCount(type: String, numberOfItems: Int) {
+            val props: MutableMap<String, String> = HashMap()
+            props["type"] = type
+            props["count"] = numberOfItems.toString()
+            AnalyticsTracker.track(Stat.READER_FOLLOWING_FETCHED, props)
+        }
+
+        @JvmStatic
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/tracker/ReaderTracker.kt:476]: If the idea of moving it to the `ViewModel` makes sense, maybe we can remove the `@JvmStatic` annotation from here and line `481`.

**Review by RenanLukas — APPROVED** [review]
Thanks for the PR, @daniloercoli ! I'm approving because it's working as expected, but I've left a couple of suggestions.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/tracker/ReaderTracker.kt (hunk 3 above)
Comment (daniloercoli) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/tracker/ReaderTracker.kt:476]: Hey @RenanLukas Thank you for your review! Your suggestions are valuable, and I will work on the PR to enhance site tracking and transfer the logic to the ViewModel within that PR.

**Pull request merged**